### ISSUE TITLE
Bash not showing Chinese right in the title

### ISSUE THREAD

**Issue by DingXiangYuanZhiXing** [microsoft/terminal#7172]
# Environment

```none
Windows build number: Microsoft Windows NT 10.0.18363.0
Windows Terminal version (if applicable): 1.1.2021.0

Any other software?
Git for windows `Git-2.28.0-64-bit.exe`
```

# Steps to reproduce

install git for windows and add bash in `settings.json` profiles as the [following stackoverflow page](https://stackoverflow.com/questions/56839307/adding-git-bash-to-the-new-windows-terminal)
```
    {
        "guid": "{00000000-0000-0000-ba54-000000000002}",
        "acrylicOpacity" : 0.75,
        "closeOnExit" : true,
        "colorScheme" : "Campbell",
        "commandline" : "\"%PROGRAMFILES%\\git\\usr\\bin\\bash.exe\" -i -l",
        "cursorColor" : "#FFFFFF",
        "cursorShape" : "bar",
        "fontFace" : "Consolas",
        "fontSize" : 10,
        "historySize" : 9001,
        "icon" : "%PROGRAMFILES%\\Git\\mingw64\\share\\git\\git-for-windows.ico",
        "name" : "Bash",
        "padding" : "0, 0, 0, 0",
        "snapOnInput" : true,
        "startingDirectory" : "%USERPROFILE%",
        "useAcrylic" : true
    },
```

# Expected behavior

The title should be right instead of meaningless letters.

# Actual behavior

The title is wrong when the path contains Chinese letters.
![image](https://user-images.githubusercontent.com/28868457/89278272-770d4780-d678-11ea-80aa-7f08e77e2da9.png)
But in the bash terminal everything seems fine. I think it's an encoding and decoding problem, but I have no idea how to resolve it.

**Comment by zadjii-msft** [comment]
I'm sorry if I'm missing this but what is the title actually supposed to be? like, what is it in the git bash window?

Is the title just being truncated? Maybe setting [`"tabWidthMode": "titleLength"`](https://docs.microsoft.com/en-us/windows/terminal/customize-settings/global-settings#tab-width-mode) might help?

**Comment by DingXiangYuanZhiXing** [comment]
Sorry, it's an encoding problem. The title is not meaningful Chinese letters.&nbsp;

----------

该邮件从移动设备发送


--------------原始邮件--------------
发件人："Mike Griese "<notifications@github.com&gt;;
发送时间：2020年8月4日(星期二) 晚上7:16
收件人："microsoft/terminal" <terminal@noreply.github.com&gt;;
抄送："DingXiangYuanZhiXing "<151120093@smail.nju.edu.cn&gt;;"Author "<author@noreply.github.com&gt;;
主题：Re: [microsoft/terminal] Bash not showing Chinese right in the title (#7172)
-----------------------------------

 

 
I'm sorry if I'm missing this but what is the title actually supposed to be? like, what is it in the git bash window?
 
Is the title just being truncated? Maybe setting "tabWidthMode": "titleLength" might help?

**Comment by DingXiangYuanZhiXing** [comment]
In the bash shell the path is correct but I don't want reveal my company name. It's not the length.

----------

该邮件从移动设备发送


--------------原始邮件--------------
发件人："Mike Griese "<notifications@github.com&gt;;
发送时间：2020年8月4日(星期二) 晚上7:16
收件人："microsoft/terminal" <terminal@noreply.github.com&gt;;
抄送："DingXiangYuanZhiXing "<151120093@smail.nju.edu.cn&gt;;"Author "<author@noreply.github.com&gt;;
主题：Re: [microsoft/terminal] Bash not showing Chinese right in the title (#7172)
-----------------------------------

 

 
I'm sorry if I'm missing this but what is the title actually supposed to be? like, what is it in the git bash window?
 
Is the title just being truncated? Maybe setting "tabWidthMode": "titleLength" might help?

**Comment by zadjii-msft** [comment]
Okay well, maybe we can try and debug a test string that's not your company's name?

In every shell, you can set the tab title manually using a command - see [this doc](https://docs.microsoft.com/en-us/windows/terminal/tutorials/tab-title#set-the-shells-title) for details. Could you try executing those commands in each of their corresponding shells with some test string that has meaningful Chinese in it, and see what the tab title ends up changing to? Maybe we can figure out if this problem is local to git bash or if it's something lower down in the stack.

**Comment by DingXiangYuanZhiXing** [comment]
@zadjii-msft I have tested in 3 shells.
* `$Host.UI.RawUI.WindowTitle = "你好，中国"` works well in powershell.
* `TITLE "你好，中国"`  works well BUT it shouldn't include double quote `""`. It should be `TITLE 你好，中国`, otherwise there will be `""` in the title like this.
![image](https://user-images.githubusercontent.com/28868457/89290912-2a336c00-d68c-11ea-848f-08bd7dd45eb3.png)
* In git bash, both `echo -ne "\033]0;你好，中国\a"` and `PS1=$
PROMPT_COMMAND=
echo -ne "\033]0;你好，中国\a"` are NOT work for Chinese. Nothing happened after the command. But it works for English letters.
![image](https://user-images.githubusercontent.com/28868457/89291088-7a123300-d68c-11ea-90d1-aca0b237b8d3.png)

**Comment by zadjii-msft** [comment]
Huh, it looks like it works in an Ubuntu tab for me: ![image](https://user-images.githubusercontent.com/18356694/89296817-85cc0e00-d628-11ea-8d17-e60228f91a54.png)

which when combined with the above, makes me think that this is a git bash specific problem. What is `$LANG` set to in git bash? (also check `LC_ALL` and `LANGUAGE`)

**Comment by DingXiangYuanZhiXing** [comment]
@zadjii-msft
![image](https://user-images.githubusercontent.com/28868457/89300418-c19fbb80-d69a-11ea-8260-3df2caaca679.png)

**Comment by zadjii-msft** [comment]
Huh, that's definitely weird. From all I can tell, this is a git bash issue, not a Terminal one. I wonder why the mintty window (the default git bash terminal) works with this? 

I guess I'll leave this open to try and debug further on our end. Thanks!

**Comment by DingXiangYuanZhiXing** [comment]
@zadjii-msft I tried in the Git Bash Mintty terminal and it works fine.

**Comment by DHowett** [comment]
I've got bad news! This works in the msys runtime shipped with git-2.27 but not in the one shipped with git-2.28. This is external.

**Comment by DHowett** [comment]
msys is no longer setting the console codepage to 65001. Evidenced by running "cmd /s /c chcp 65001" before setting the title.
